Note that files in /run/secrets/ can only be copied using RUN cp (#1400)
balena-io/docs: shared/deployment/secrets/_default.md

               dest: my-recipe2
```

This will mount the `super-secret-recipe` file into `/run/secrets/my-recipe` file in
every build container. However, the `/run/secrets` folder will not be added to the

> "build context" (tar stream) that is sent to the Docker daemon (or balenaEngine),

Reviewer: ```suggestion
build context that is sent to the Docker daemon (or balenaEngine),
```

I'd remove quotes as build context is consistent with terminology we have used elsewhere.

and therefore:

* The `/run/secrets/` folder will not be present in the image that is deployed to the devices.
* The `COPY` or `ADD` Dockerfile directives will **not** be able to copy files from that folder.
* The folder will be available "for the duration of `RUN` directives", such that secrets can be
  accessed by scripts/code at image build time, for example:

> ```sh

Reviewer: ```suggestion
```shell
```

I know this is an alias but for consistency let's update to shell as is used elsewhere.

               dest: my-recipe2
```

This will mount the `super-secret-recipe` file into `/run/secrets/my-recipe` file in
every build container. However, the `/run/secrets` folder will not be added to the
build context that is sent to the Docker daemon (or balenaEngine),

> and therefore:

Reviewer 2: Oh wait -- wasn't this supposed to have replace the paragraph in line 42? (Rather than replacing the paragraph in line 75...)

This will mount the `super-secret-recipe` file as `/run/secrets/my-recipe` for all services, and `super-secret-recipe-2` as `/run/secrets/my-recipe2` for `service1` only. Again, note that the `/run/secrets` folder is only available during the image build, and not present in the image that is deployed to the devices.

Subdirectories are supported in both the source (`.balena/secrets`) and the destination (`/run/secrets`).

> It is worth mentioning that if you want to copy the file `super-secret-recipe` from the `/run/secrets/my-recipe` folder to the container you must use the RUN directive, e.g. `RUN cp /run/secrets/my-recipe <destination>`.

Reviewer 2: How about this: there was already a paragraph earlier in this document that touched the topic:
```
This will mount the `super-secret-recipe` file into `/run/secrets/my-recipe` file in
every build container. Note that the `/run/secrets` folder is only available during
the image build, and not present in the image that is deployed to the devices.
```

I suggest amending that paragraph instead of adding a new paragraph here. Amending it like:

```
This will mount the `super-secret-recipe` file into `/run/secrets/my-recipe` file in
every build container. However, the `/run/secrets` folder will not be added to the
"build context" (tar stream) that is sent to the Docker daemon (or balenaEngine),
and therefore:

* The `/run/secrets/` folder will not be present in the image that is deployed to the devices.
* The `COPY` or `ADD` Dockerfile directives will **not** be able to copy files from that folder.
* The folder will be available "for the duration of `RUN` directives", such that secrets can be
  accessed for example by scripts/code at image build time:

` ``sh
RUN cat /run/secrets/my-recipe | secret_processing_script.sh
` ``
```
The `RUN cat` example above is probably less controversial than `RUN cp`. :-)

Reviewer: @OdysLam the above clarification from Paulo looks great. Do you agree and want to update this PR to include?

Author: Totally onboard @garethtdavies. Thank you @pdcastro for putting the time for these changes. 

Will update the PR today.

> It is worth mentioning that if you want to copy the file `super-secret-recipe` from the `/run/secrets/my-recipy` folder to the container, you can't use the Dockerfile primitive `COPY`, but rather you have to use `RUN cp /run/secrets/my-recipy <destination>`.

Reviewer 3: ```suggestion
It is worth mentioning that if you want to copy the file `super-secret-recipe` from the `/run/secrets/my-recipe` folder to the container you must use the RUN directive, e.g. `RUN cp /run/secrets/my-recipe <destination>`.
```
